thread.c

```diff
@@ -3362,18 +3362,18 @@ threadptr_local_aset(rb_thread_t *th, ID id, VALUE val)
 	return val;
     }
     else {
-	st_table *local_storage = th->ec->local_storage;
+	struct rb_id_table *local_storage = th->ec->local_storage;
 
 	if (NIL_P(val)) {
 	    if (!local_storage) return Qnil;
-	    st_delete_wrap(local_storage, id);
+	    rb_id_table_delete(local_storage, id);
 	    return Qnil;
 	}
 	else {
 	    if (local_storage == NULL) {
-		th->ec->local_storage = local_storage = st_init_numtable();
+		th->ec->local_storage = local_storage = rb_id_table_create(0);
```

Comment:
What is 0 here?

Reply:
The size of the ID table, 0 is the minimum but would allocate a capacity for 4 entries https://github.com/ruby/ruby/blob/a515e3f99ea50358afbfc94a53d54a390096626d/id_table.c#L72

See https://github.com/ruby/ruby/pull/2083 which can probably be resurrected as well.

Reply:
Yes, #2083 should be addressed next.